runtime/smithy-client/common/src/aws/smithy/kotlin/runtime/client/config/HttpChecksumClientConfig.kt

```diff
@@ -0,0 +1,40 @@
+package aws.smithy.kotlin.runtime.client.config
+
+/**
+ * Client config for HTTP checksums
+ */
+public interface HttpChecksumClientConfig {
+    /**
+     * Configures request checksum calculation
+     */
+    public val requestChecksumCalculation: HttpChecksumConfigOption?
+
+    /**
+     * Configures response checksum validation
+     */
+    public val responseChecksumValidation: HttpChecksumConfigOption?
+
+    public interface Builder {
+        /**
+         * Configures request checksum calculation
+         */
+        public var requestChecksumCalculation: HttpChecksumConfigOption?
+
+        /**
+         * Configures response checksum validation
+         */
+        public var responseChecksumValidation: HttpChecksumConfigOption?
+    }
+}
+
+public enum class HttpChecksumConfigOption {
```

Comment:
correctness: Right now request and response checksum configs both use this enum. In the future, their config options might diverge, which will force us to make an API break. I'd recommend splitting this into `HttpChecksumRequestConfigOption` and `HttpChecksumResponseConfigOption` now to get ahead of that.

Reply:
It seems like just speculation for now, I say we only separate them when that time comes, changing the type of this config option would be a breaking change for all SDKs. I think it's more likely that we would make additive changes

Reply:
It's not speculative, there was some discussion during the spec review about adding more options. 

The spec calls out `RequestChecksumCalculation` and `ResponseChecksumValidation` as separate config types, so adding a new config option to either of those wouldn't be a breaking change for SDKs which implement it as the spec says.